RELEASING.md

```diff
@@ -18,4 +18,6 @@ Builds and releases are automated with GitHub Actions and GoReleaser. There are
    
    The action creates a "draft" release. Go to [Releases](https://github.com/canonical/terraform-provider-maas/releases) to open it, select `edit`, and select `Publish release` if you are happy. 
 
-3. Verify the published release looks good by reviewing [Releases](https://github.com/canonical/terraform-provider-maas/releases).
+3. Verify the release is published by: 
+   1. Checking the release is now the latest published under [Releases](https://github.com/canonical/terraform-provider-maas/releases). 
+   2. Checking the [Hashicorp Registry provider page](https://registry.terraform.io/providers/canonical/maas/latest) is displaying the released version as latest. This could take approximately 30 minutes to update.
```

Comment:
nit
```suggestion
   2. Checking the [HashiCorp Registry provider page](https://registry.terraform.io/providers/canonical/maas/latest) is displaying the released version as latest. This could take approximately 30 minutes to update.
```